### PR TITLE
feat: Created Webhook parser for Mandrill

### PR DESCRIPTION
**Pull request by abhilipsasahoo03**
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Added Webhook parser for Mandril (email)
- **Why this change was needed?** (You can also link to an open issue here)
Fixes Issue #1447
- **Other information**:
For example of Mandrill Webhook JSON data, refer: https://gist.github.com/ramonsmits/7563502
For official docs, refer: https://mailchimp.com/developer/transactional/docs/webhooks/

### CODE REVIEW

**Comment by abhilipsasahoo03** [comment]
@scopsy @davidsoderberg could you please take a look?

File: providers/mandrill/src/lib/mandrill.provider.ts [hunk 1]
```diff
@@ -73,4 +75,55 @@ export class MandrillProvider implements IEmailProvider {
...
+      case 'open':
+        return EmailEventStatusEnum.OPENED;
+      case 'hard_bounce':
+        return EmailEventStatusEnum.BOUNCED;
+      case 'click':
+        return EmailEventStatusEnum.CLICKED;
+      case 'send':
+        return EmailEventStatusEnum.SENT;
+      case 'spam':
+        return EmailEventStatusEnum.SPAM;
```
Comment (p-fernandez) [inline providers/mandrill/src/lib/mandrill.provider.ts:126]: Creating a `MandrillStatusEnum` enum it would be the icing in the cake for this PR.

Comment (p-fernandez) [reply providers/mandrill/src/lib/mandrill.provider.ts:126]: For now I think it can safely live in `mandrill.provider.ts`.

**Comment by abhilipsasahoo03** [comment]
@p-fernandez please take a look and let me know if anything else is required

File: providers/mandrill/src/lib/mandrill.provider.ts [hunk 2]
```diff
@@ -5,11 +5,25 @@ import {
...
+  OPEN = 'opened',
+  SEND = 'sent',
+  DEFERRAL = 'deferred',
+  HARD_BOUNCE = 'hard_bounced',
+  SOFT_BOUNCE = 'soft_bounced',
+  CLICK = 'clicked',
+  SPAM = 'spam',
+  UNSUB = 'unsubscribed',
+  REJECT = 'rejected',
+}
```
Comment (p-fernandez) [inline providers/mandrill/src/lib/mandrill.provider.ts:25]: You should keep the status naming in participles, for a meaning thing as they are status not actions.
```suggestion
export enum MandrillStatusEnum {
  OPENED = 'opened',
  SENT = 'sent',
  SOFT_BOUNCED = 'soft_bounced',
  CLICKED = 'clicked',
  REJECTED = 'rejected',
  ...
}
```

File: providers/mandrill/src/lib/mandrill.provider.ts [hunk 3]
```diff
@@ -81,4 +95,63 @@ export class MandrillProvider implements IEmailProvider {
...
+      case 'reject':
+        return MandrillStatusEnum.REJECT;
+      case 'soft_bounce':
+        return MandrillStatusEnum.SOFT_BOUNCE;
+      case 'unsub':
+        return MandrillStatusEnum.UNSUB;
+      case 'deferral':
+        return MandrillStatusEnum.DEFERRAL;
+    }
+  }
```
Comment (p-fernandez) [inline providers/mandrill/src/lib/mandrill.provider.ts:156]: You swapped the wrong thing. The cases are the ones that you need to use the `MandrillStatusEnum` with, and return a status of `EmailStatusEnum` that is the internal of Novu, therefore doing the mapping needed between the 2 systems.

**Review by p-fernandez — CHANGES_REQUESTED** [review]

File: providers/mandrill/src/lib/mandrill.provider.ts (hunk 3 above)
Comment (abhilipsasahoo03) [reply providers/mandrill/src/lib/mandrill.provider.ts:156]: But what status should `EmailEventStatusEnum` should return in case of deferral, or

**Comment by abhilipsasahoo03** [comment]
@p-fernandez please take a look and check if this is what you suggested. Let me know if anything else is required.

**Comment by abhilipsasahoo03** [comment]
@p-fernandez @davidsoderberg @scopsy could you please let me know why Test providers / test is failing.

**Comment by abhilipsasahoo03** [comment]
@p-fernandez please take a look

**Comment by abhilipsasahoo03** [comment]
@p-fernandez @scopsy @davidsoderberg kindly review.

**Pull request merged**